### ISSUE TITLE
Part of model shadow disappears when motion is loaded

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-gui#479]
After opening model:
![](https://user-images.githubusercontent.com/4203505/34846695-d81a4704-f6cd-11e7-98c4-24810199675a.png)

After loading motion:
![](https://user-images.githubusercontent.com/4203505/34846700-dcf7fb72-f6cd-11e7-9148-c620da62c0e6.png)

**Comment by aymanhab** [comment]
Likely reason is that the motion moves the model so it's outside the "cone" of the light source. Please check the placement of the light source to see if that explains. Thanks

**Comment by tkuchida** [comment]
You're right: the model moved but the light source did not. Would it make sense to anchor the light to the model rather than ground? Also, I happened to load a motion from treadmill walking, but if the subject (model) walks across the lab (screen), I think the shadow will end up getting clipped sometime during the motion.

**Comment by aymanhab** [comment]
Thanks for checking @tkuchida There's a scene light attached to world ground (that causes shadows) and there is actually model point-light attached to the model's Ground (causes no shadows). These can be rearranged.  Motions in general do not move the Ground but move the model relative o its ground. IF we have a concept of base segment attached to ground then we can use thta instead and the model light would move.  Thoughts?

**Comment by tkuchida** [comment]
I don't know the best approach, but I think a user would want it to *just work*—that is, the GUI should do something automatically so that, by default, there are no clipped shadows. I think the user should need to adjust the position of the light only if they want to do something fancy; it should not be necessary as part of a typical workflow. Here's one idea:

- User opens model. Scene light is moved to cast shadow on model, as is done currently.
- User loads motion. Scene light is anchored to model before modifying pose (i.e., the scene light moves with the model) and is then re-anchored to ground.
- User presses "play" button. Camera and scene light remain fixed; model walks away from camera and light (shadow elongates as model walks away and out of frame).
- User presses a "follow model" toggle button and presses "play" again. Camera and scene light are now anchored to the model (fixed in the model's ground frame). Might need to set the position of the camera using a moving average of the model's origin to avoid camera bounce/jitter.
